src/StreamJsonRpc.Tests/JsonRpcWithFatalExceptionsTests.cs

```diff
@@ -170,6 +170,50 @@ public async Task CancelMayStillReturnErrorFromServer()
         Assert.False(this.serverRpc.IsDisposed);
     }
 
+    [Fact]
+    public async Task DisposedClientResultsInCancelled()
+    {
+        using (var cts = new CancellationTokenSource())
+        {
+            var invokeTask = this.clientRpc.InvokeWithCancellationAsync<string>(nameof(Server.AsyncMethodFaultsAfterCancellation), new[] { "a" }, cts.Token);
+            await this.server.ServerMethodReached.WaitAsync(this.TimeoutToken);
+            this.clientRpc.Dispose();
+            this.server.AllowServerMethodToReturn.Set();
+
+            // Connection was closed before error was sent from the server
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(() => invokeTask);
+            Assert.Equal(0, this.serverRpc.IsFatalExceptionCount);
+        }
+
+        Assert.True(((IDisposableObservable)this.clientMessageHandler).IsDisposed);
+        Assert.True(((IDisposableObservable)this.serverMessageHandler).IsDisposed);
+        Assert.True(this.clientRpc.IsDisposed);
+        Assert.False(this.serverRpc.IsDisposed);
+    }
+
+    [Fact]
+    public async Task UnexpectedDisconnectResultsInConnectionLostException()
```

Comment:
I'm glad you included this test, but it isn't really about fatal server exceptions--just about the server hanging up. Can you move this to `JsonRpcTests.cs` and simplify (e.g. remove the cts, dispose of `serverRpc` itself, etc.)?

Reply:
In fact we have other tests that verify this (e.g. `CloseStreamsOnSynchronousMethodException`) so I think we can just remove it.

Reply:
I removed this.